src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationResolver.cs

```diff
@@ -0,0 +1,269 @@
+                    {
+                        foreach (var languageWorkerPath in Directory.EnumerateDirectories(probingPath))
+                        {
+                            string languageWorkerDir = Path.GetFileName(languageWorkerPath).ToLower();
+
+                            // If probing paths are malformed and have duplicate directories of the same language worker (eg. due to different casing)
+                            if (outputDict.ContainsKey(languageWorkerDir))
+                            {
+                                continue;
+                            }
+
+                            // Only skip worker directories that don't match the current runtime or are not enabled via hosting config
+                            // Do not skip non-worker directories like the function app payload directory
+                            if (languageWorkerPath.StartsWith(fallbackPath) || languageWorkerPath.StartsWith(probingPath))
+                            {
+                                if ((_workersAvailableForResolutionViaHostingConfig is not null &&
+                                    !_workersAvailableForResolutionViaHostingConfig.Contains(languageWorkerDir)) ||
+                                    (!_environment.IsMultiLanguageRuntimeEnvironment() &&
+                                    !_environment.IsPlaceholderModeEnabled() &&
+                                    workerRuntime is not null &&
+                                    !workerRuntime.Equals(languageWorkerDir, StringComparison.OrdinalIgnoreCase)))
+                                {
+                                    continue;
+                                }
+                            }
+
+                            IEnumerable<string> workerVersions = Directory.EnumerateDirectories(languageWorkerPath);
+                            var versions = ParseWorkerVersionsInDescending(workerVersions);
+
+                            GetWorkerConfigsFromProbingPaths(versions, languageWorkerPath, languageWorkerDir, releaseChannel, outputDict);
+                        }
+                    }
+                }
+            }
+
+            if (!_environment.IsMultiLanguageRuntimeEnvironment() &&
+                workerRuntime is not null &&
+                outputDict.ContainsKey(workerRuntime))
+            {
+                return outputDict.Values.ToList();
+            }
+
+            _logger.LogDebug("Searching for worker configs in the fallback directory.");
+
+            // Search in fallback path if worker cannot be found in probing paths
+            GetWorkerConfigsFromWithinHost(fallbackPath, workerRuntime, outputDict);
+
+            return outputDict.Values.ToList();
+        }
+
+        private void GetWorkerConfigsFromProbingPaths(IEnumerable<Version> versions, string languageWorkerPath, string languageWorkerFolder, string releaseChannel, ConcurrentDictionary<string, string> outputDict)
```

Comment:
The Get prefix suggests that this method returns something but looks like it returns void. May be update the method name?